include/highfive/bits/H5Node_traits_misc.hpp

```diff
@@ -162,6 +162,42 @@ inline std::string NodeTraits<Derivate>::getObjectName(size_t index) const {
     return std::string(bigBuffer.data(), length);
 }
 
+template <typename Derivate>
+inline std::string NodeTraits<Derivate>::getPath() const {
```

Comment:
The code here seems to be largely identical with the above, could you please remind me why this could not be simplified?

Reply:
It's more about avoiding repeated code. Also, if you read the same pattern several times in the code, you wonder if there is any difference, and why… so I for example spend more energy doing mental diffs to see if there's anything special in similar looking code…

I have an example locally, just thinking about ways to make it more intuitive before I push it somewhere.

Reply:
See [this commit](https://github.com/BlueBrain/HighFive/commit/fedd857059a21ce43ed263bc3ee4da0f685bb9c0). It's somewhat hideous, but so is the nature of C++, and the fact that HDF5 has two pretty much identical functions (in purpose) with arguments swapped in the signature.